### PR TITLE
cppcheck: fix some memleaks

### PR DESCRIPTION
**Pull request by serval2412**
[fontforge/python.c:8017]: (error) Memory leak: ret
[fontforge/splinefill.c:1535]: (error) Memory leak: sum
[fontforge/splineutil.c:5791]: (error) Memory leak: pat
[fontforge/splineutil.c:5793]: (error) Memory leak: pat
[fontforge/splineutil.c:5813]: (error) Memory leak: grad
[fontforge/splineutil.c:5815]: (error) Memory leak: grad
[fontforge/tottfvar.c:67]: (error) Memory leak: ss
[fontforge/tottfvar.c:67]: (error) Memory leak: sp
[fontforge/ttfspecial.c:2208]: (error) Memory leak: bdfinfo
[fontforge/winfonts.c:917]: (error) Memory leak: file_lens
[fontforgeexe/acorn2sfd.c:616]: (error) Memory leak: mapping
[fontforgeexe/basedlg.c:516]: (error) Memory leak: mapping
[gdraw/gimageclut.c:507]: (error) Memory leak: clutinf

### CODE REVIEW

File: fontforge/tottfvar.c [hunk 1]
```diff
@@ -63,14 +63,24 @@ static int AssignPtNumbers(MMSet *mm,int gid) {
 	for ( i=0; i<=mm->instance_count; ++i )
 	    if ( ss[i]!=NULL ) stillmore = true;
 	if ( stillmore )
```
Comment (jtanx) [inline fontforge/tottfvar.c]: This can be simplified to move the free statements before `if ( stillmore )`

Comment (serval2412) [inline fontforge/tottfvar.c]: As you may have seen, I've updated the pull request.

File: fontforgeexe/acorn2sfd.c [hunk 2]
```diff
@@ -562,6 +562,8 @@ return;
 	    free(widths);
 	}
     }
+    if (m != 0)
+        free(mapping);
```
Comment (frank-trampe) [inline fontforgeexe/acorn2sfd.c]: It's not worth rewriting the pull request, but I consider it generally good form to set the pointer to NULL after freeing if the function is not terminating immediately. It makes detecting errors from use of freed memory easier to identify.

**Comment by frank-trampe** [comment]
Looks good to me.

File: fontforgeexe/acorn2sfd.c (hunk 2 above)
Comment (serval2412) [inline fontforgeexe/acorn2sfd.c]: Now I found how to edit directly a commit, no problem for this :-)
(great function!!)

Comment (frank-trampe) [inline fontforgeexe/acorn2sfd.c]: Too late. I already merged it.

**Pull request merged**